tests/unit/mainwindow/test_prompt.py

```diff
@@ -130,3 +131,14 @@ def test_root_path(self, get_prompt):
         """With / as path, show root contents."""
         prompt = get_prompt('/')
         assert prompt._file_model.rootPath() == '/'
+    
+    def test_expand_user(self, qtbot, get_prompt):
+        prompt = get_prompt('/')
+        prompt._lineedit.setText('')
+        qtbot.keyPress(prompt._lineedit, Qt.Key_AsciiTilde)
+        # The path currently seen by the file_model should be the same as os.path.expanduser('~')
+        assert str(Path(prompt._file_model.rootPath()) / prompt._to_complete) == os.path.expanduser('~')
+        # The first character on the lineedit should remain ~
+        prompt.item_focus('next')
+        prompt.item_focus('next')
```

Comment:
What happens here if e.g. the home directory is empty?

Reply:
Yeah maybe I should make a couple directories in this test. Is that okay to do considering this is using the home directory?

Reply:
You can use pytest's [tmp_path fixture](https://docs.pytest.org/en/6.2.x/tmpdir.html) plus [monkeypatch](https://docs.pytest.org/en/6.2.x/monkeypatch.html) to patch it out. See `tests/unit/completion/test_models.py` for [an example](https://github.com/qutebrowser/qutebrowser/blob/master/tests/unit/completion/test_models.py#L427-L430).